src/ape/api/accounts.py

```diff
@@ -173,16 +173,16 @@ def deploy(self, contract: "ContractContainer", *args, **kwargs) -> "ContractIns
         if not receipt.contract_address:
             raise AccountsError(f"'{receipt.txn_hash}' did not create a contract.")
 
-        address = click.style(receipt.contract_address, bold=True)
+        address = receipt.contract_address
+        styled_address = click.style(receipt.contract_address, bold=True)
         contract_name = contract.contract_type.name or "<Unnamed Contract>"
-        logger.success(f"Contract '{contract_name}' deployed to: {address}")
-
-        contract_instance = self.create_contract(
-            address=receipt.contract_address,  # type: ignore
-            contract_type=contract.contract_type,
+        logger.success(f"Contract '{contract_name}' deployed to: {styled_address}")
+        contract_instance = self.chain_manager.contracts.instance_at(
+            address, contract.contract_type
         )
-        self.chain_manager.contracts[contract_instance.address] = contract_instance.contract_type
-        return contract_instance
+        # NOTE: Type ignore because we know it is a contract instance at this point
```

Comment:
Could throw a `assert contract_instance` in here to satisfy mypy

Reply:
mypy is mad because the type is BaseAddress and not ContractInstance

Normally I would check the type to satisfy mypy (or assert the type), but importing from the `ape.contracts` namespace is challenging (circular imports)

Reply:
how about `assert not isinstance(contract_instance, AddressType)`?

Reply:
It would be `BaseAddress`, but let me try

Reply:
Oh no that won't work because `ContractInstance` is a subclass of `BaseAddress`

Reply:
TBH I kind of wish `instance_at()` always returned a `ContractInstance`. I don't remember why we made it return `Address` when it does not find the contract

Reply:
> TBH I kind of wish `instance_at()` always returned a `ContractInstance`. I don't remember why we made it return `Address` when it does not find the contract

I don't either. Let's change that? Or have it return `Optional[ContractInstance]` maybe

Reply:
Or maybe `ape.Contract` is more of the API to access the contract cache?

Reply:
ape.Contract calls instance_at() at under-the-hood and manager-access mixin impls use chain_manager.instance_at directly since they can't really import from the top-level namespace.

Reply:
See all the changes in the latest commit! Making this change makes things way  easier and still fixes #777, as now things will fail loudly and cause alert.